File: src/main/java/com/zhihao/admin/module/system/controller/OperLogController.java
package com.zhihao.admin.module.system.controller;

import com.zhihao.admin.common.api.ApiResponse;
import com.zhihao.admin.common.api.PageQuery;
import com.zhihao.admin.common.api.PageResult;
import com.zhihao.admin.module.system.entity.SysOperLog;
import com.zhihao.admin.module.system.service.OperLogService;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.security.access.prepost.PreAuthorize;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/system/operation-logs")
@RequiredArgsConstructor
public class OperLogController {

    private final OperLogService operLogService;

    @GetMapping
    @PreAuthorize("hasAuthority('system:operlog:list')")
    public ApiResponse<PageResult<SysOperLog>> page(@Valid PageQuery query) {
        return ApiResponse.ok(operLogService.page(query));
    }
}

File: nest-server/src/modules/oper-log/oper-log.controller.ts
import { Controller, Get, Query, UseGuards } from '@nestjs/common'
import { ok } from '../../common/api-response'
import { pageQuerySchema } from '../../common/page'
import { PermissionGuard } from '../../common/permission.guard'
import { RequireAuthority } from '../../common/authority.decorator'
import { OperLogService } from './oper-log.service'

@Controller('api/system/operation-logs')
@UseGuards(PermissionGuard)
export class OperLogController {
  constructor(private readonly operLogService: OperLogService) {}

  @Get()
  @RequireAuthority('system:operlog:list')
  async page(@Query() query: unknown) {
    const data = pageQuerySchema.parse(query)
    return ok(await this.operLogService.page(data))
  }
}
